### PR TITLE
New reglementen view

### PR DESCRIPTION
**Pull request by lagartoverde**
## Overview
Implemented new reglementen view based on the designs

##### connected issues and PRs:
https://binnenland.atlassian.net/browse/GN-5976
https://github.com/lblod/besluit-publicatie-publish-service/pull/16
https://github.com/lblod/app-gn-publicatie/pull/44

### Setup
Made sure you are running the correct branch of the backend, and the besluits are imported using the new version of the publish service

### How to test/reproduce
Publish a few chains of besluits (you can even produce trees for edge cases)
Go to the new reglementen view and verify that everything works as expected and according to the design

### Challenges/uncertainties
None

### Checks PR readiness
- [x] UI: works on smaller screen sizes
- [x] UI: feedback for any loading/error states
- [x] Check cancel/go-back flows
- [x] Check database state correct when deleting/updating (especially regarding relationships)
- [x] changelog
- [x] npm lint
- [x] no new deprecations

### CODE REVIEW

**Review by abeforgit — CHANGES_REQUESTED** [review]
Looking great at first glance, but noticed a small bug, when the search has no results, it throws an error

Image

File: app/routes/bestuurseenheid/reglementen/index.js [hunk 1]
```diff
@@ -84,7 +111,7 @@ export default class BestuurseenheidReglementenIndexRoute extends Route {
         ?adminUnit <http://data.vlaanderen.be/ns/mandaat#isTijdspecialisatieVan> ?adminUnitGeneral.
         ?adminUnitGeneral besluit:bestuurt ${sparqlEscapeUri(
           bestuurseenheid.uri
-        )}
+        )}.
         VALUES ?besluitType { ${DECISION_TYPES_TO_LINK.map(
```
Comment (abeforgit) [inline app/routes/bestuurseenheid/reglementen/index.js]: putting this at the start of the query might already help. the order of statements can make a big difference for query performance

File: app/routes/bestuurseenheid/reglementen/index.js [hunk 2]
```diff
@@ -84,7 +111,7 @@ export default class BestuurseenheidReglementenIndexRoute extends Route {
         ?adminUnit <http://data.vlaanderen.be/ns/mandaat#isTijdspecialisatieVan> ?adminUnitGeneral.
```
Comment (abeforgit) [inline app/routes/bestuurseenheid/reglementen/index.js]: i'd make this one prefixed as well for readability

File: app/routes/bestuurseenheid/reglementen/index.js [hunk 3]
```diff
@@ -84,7 +111,7 @@ export default class BestuurseenheidReglementenIndexRoute extends Route {
...
         ?adminUnitGeneral besluit:bestuurt ${sparqlEscapeUri(
```
Comment (abeforgit) [inline app/routes/bestuurseenheid/reglementen/index.js]: I'd also put this more to the top, I think limiting as much as possible before you start filling variables is the way to go. Depending on how good the query optimizer is in virtuoso, this might not matter, but it shouldn't hurt

File: app/routes/bestuurseenheid/reglementen/index.js
Comment (abeforgit) [inline app/routes/bestuurseenheid/reglementen/index.js]: Also here swap the filters, the searchfilter is harder to calculate than the existence of a triple, so if we can run it on fewer results that should be better

Comment (abeforgit) [inline app/routes/bestuurseenheid/reglementen/index.js:146]: this looks weird to me, a triple pattern block with only a filter in it, is that intentional?

**Review by abeforgit — APPROVED** [review]
problems are fixed, some ideas for maybe improving the query speed

File: app/routes/bestuurseenheid/reglementen/index.js
Comment (lagartoverde) [reply app/routes/bestuurseenheid/reglementen/index.js:146]: My logic is, I want to get either a decision linked to the original decision or a decision that doesn't have a child decision so it's a original besluit

File: app/templates/bestuurseenheid/reglementen/history.hbs [hunk 4]
```diff
@@ -0,0 +1,57 @@
+{{page-title
+  (concat
+    @model.bestuurseenheid.classificatie.label ' ' @model.bestuurseenheid.naam
+  )
+}}
+<div class="flex-one">
```
Comment (piemonkey) [inline app/templates/bestuurseenheid/reglementen/history.hbs]: I know it's not in the design, but I think this template needs a `breadcrumb` entry, as otherwise it's not clear how to get back from the `historiek` page.

**Review by piemonkey — COMMENTED** [review]
Looks good. I don't know enough about optimising SPARQL to comment on that part though.

Is it expected that publishing 2 APs, each from the same 'original version' leads to 2 entries in the reglementen list, one for each of the 'child' APs, and each not including the other in the history? It makes sense to me to have 2 versions, but I'm not sure if they should all appear in the history or not.

To be clear, what I'm talking about is, I have 3 published APs:
- AP original
- AP 2 (original version AP original)
- AP 3 (original version AP original)
I then get 2 entries in reglementen, one for AP 2, with history of AP 2 -> AP original and the other for AP 3 with history AP 3 -> AP original.

**Comment by abeforgit** [comment]
> Looks good. I don't know enough about optimising SPARQL to comment on that part though.
> 
> Is it expected that publishing 2 APs, each from the same 'original version' leads to 2 entries in the reglementen list, one for each of the 'child' APs, and each not including the other in the history? It makes sense to me to have 2 versions, but I'm not sure if they should all appear in the history or not.
> 
> To be clear, what I'm talking about is, I have 3 published APs:
> 
> * AP original
> * AP 2 (original version AP original)
> * AP 3 (original version AP original)
>   I then get 2 entries in reglementen, one for AP 2, with history of AP 2 -> AP original and the other for AP 3 with history AP 3 -> AP original.

It's expected yeah, at least in the sense that that's the best we can do for now in the case of a tree-shaped history
We expect/hope that to be an exceptional case

File: app/utils/sparql.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  }
+}
+/**
+ *
+ * Converts a sparql binding to a simple object contain a mapping of the binding keys to their values
+ */
+export function bindingToObject(binding) {
+  return Object.fromEntries(
+    Object.entries(binding).map(([key, term]) => [key, term.value])
+    // TS doesn't give us 'keyof' from Object.entries() as a subclass could have extra fields,
```
Comment (abeforgit) [inline app/utils/sparql.js]: this is a js file, and you're not putting it through zod...

File: app/routes/bestuurseenheid/reglementen/index.js [hunk 6]
```diff
@@ -31,20 +80,134 @@ export default class BestuurseenheidReglementenIndexRoute extends Route {
...
+        (a, b) => uittrekselIds.indexOf(b.id) - uittrekselIds.indexOf(a.id)
+      );
+    } else {
+      uittrekselsSync = [];
+    }
+
+    uittrekselsSync.meta = generateMeta(params, count);
+    return { bestuurseenheid, uittreksels: uittrekselsSync };
+  }
+  buildSort(sort) {
```
Comment (abeforgit) [inline app/routes/bestuurseenheid/reglementen/index.js]: a lot of this is duplicated code from the history route...

File: app/routes/bestuurseenheid/reglementen/index.js [hunk 7]
```diff
@@ -31,20 +80,134 @@ export default class BestuurseenheidReglementenIndexRoute extends Route {
...
+    const prefixes = `
+      PREFIX besluit: <http://data.vlaanderen.be/ns/besluit#>
+      PREFIX eli: <http://data.europa.eu/eli/ontology#>
+      PREFIX prov: <http://www.w3.org/ns/prov#>
+      PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+      PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+      PREFIX dct: <http://purl.org/dc/terms/>
+      PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+    `;
+    const queryContent = `
```
Comment (abeforgit) [inline app/routes/bestuurseenheid/reglementen/index.js:106]: these queries (also the one in the history route) could use some comment to explain what's going on

personally I'd extract them to their own file as well

**Review by abeforgit — CHANGES_REQUESTED** [review]

File: app/routes/bestuurseenheid/reglementen/index.js
Comment (abeforgit) [reply app/routes/bestuurseenheid/reglementen/index.js:146]: ok I guess that makes sense since `?uri` should already be constrained to a decision type from higher up

File: app/routes/bestuurseenheid/reglementen/index.js (hunk 7 above)
Comment (lagartoverde) [reply app/routes/bestuurseenheid/reglementen/index.js:106]: I didn't extract them because I think that they are too bounded to the route that they are not going to be useful for anything else.
I added some comments so they are more clear

**Comment by lagartoverde** [comment]
Dedicated more time to it and solved the problems with trees. Some parts of the query might be a little bit weird but that was what it was faster while testing, still open to suggestions about improvements but quite happy how it turned out :)
The query is super fast and takes care of all the leafs

File: app/templates/bestuurseenheid.hbs [hunk 8]
```diff
@@ -27,8 +27,10 @@
   <m.sidebar>
     <Sidebar />
   </m.sidebar>
-  <m.content @scroll={{true}}>
-    {{outlet}}
-    <Footer/>
+  <m.content>
```
Comment (piemonkey) [inline app/templates/bestuurseenheid.hbs:30]: It seems that this no longer scrolls? Only when the screen is wide enough that we see the sidebar on the left.

**Review by piemonkey — CHANGES_REQUESTED** [review]
My other comments look good, but the scrolling for meetings is broken

**Review by piemonkey — APPROVED** [review]
Looks good now!
